source/Detector/Amptek/AmptekEventDefinitions.cpp

```diff
@@ -0,0 +1,193 @@
+		mcaSource_ = other.mcaSource_;
+		preampState_ = other.preampState_;
+		peakDetectMode_ = other.peakDetectMode_;
+		pileUpRejection_ = other.pileUpRejection_;
+		coolerSetting_ = other.coolerSetting_;
+		scaIndices_ = other.scaIndices_;
+		scaHighIndices_ = other.scaHighIndices_;
+		scaLowIndices_ = other.scaLowIndices_;
+		fastThreshold_ = other.fastThreshold_;
+		slowThreshold_ = other.slowThreshold_;
+		peakingTime_ = other.peakingTime_;
+		fastChannelPeakingTime_ = other.fastChannelPeakingTime_;
+	}
+	return *this;
+}
+
+void AmptekConfigurationData::parseConfiguration(QString configuration)
+{
+	QString configurationValue = configuration.section('=', -1);
+
+	if(configuration.contains("GAIA"))
+		analogGainIndex_ = configurationValue.toInt();
+	else if(configuration.contains("GAIF"))
+		fineGain_ = configurationValue.toDouble();
+	else if(configuration.contains("GAIN"))
+		totalGain_ = configurationValue.toDouble();
+	else if(configuration.contains("HVSE"))
+		hvSet_ = configurationValue;
+	else if(configuration.contains("MCAC"))
+		mcaCount_ = configurationValue.toInt();
+	else if(configuration.contains("MCAE"))
+		mcaEnabled_ = configurationValue;
+	else if(configuration.contains("MCAS"))
+		mcaSource_ = configurationValue;
+	else if(configuration.contains("PAPS"))
+		preampState_ = configurationValue;
+	else if(configuration.contains("PDMD"))
+		peakDetectMode_ = configurationValue;
+	else if(configuration.contains("PURE"))
+		pileUpRejection_ = configurationValue;
+	else if(configuration.contains("SCAI"))
+		scaIndices_.append(configurationValue.toInt());
+//TODO: figure out the definitions of these two configurations...
+//	else if (configuration.contains("HightIncides"))
+//		scaHighIndices_.append(configurationValue.toInt());
+//	else if (configuration.contains("LowIncides"))
+//		scaLowIndices_.append(configurationValue.toInt());
+	else if(configuration.contains("SCAH")){
+		//ignore this
+	}
+	else if(configuration.contains("SCAL")){
+		//ignore this
+	}
+	else if(configuration.contains("TECS"))
+		coolerSetting_ = configurationValue;
+	else if(configuration.contains("THFA"))
+		fastThreshold_ = configurationValue.toDouble();
+	else if(configuration.contains("THSL"))
+		slowThreshold_ = configurationValue.toDouble();
+	else if(configuration.contains("TPEA"))
+		peakingTime_ = configurationValue.toDouble();
+	else if(configuration.contains("TPFA"))
+		fastChannelPeakingTime_ = configurationValue.toInt();
+}
+
+
+/// =============== implementation of AmptekConfigurationValuesEvent ================
+void AmptekConfigurationValuesEvent::parseConfigurations(QStringList configurationList)
+{
+	for(int x = 0, size = configurationList.count(); x < size; x++){
+		QString configuration = configurationList.at(x);
+		QString configurationValue = configuration.section('=', -1);
+// TODO: after figuring out the defintions of configurationValueHightIndices / configurationValueLowIndices, we can use this function
```

Comment:
When the correct configuration strings are figured out, this will removed and the commented codes will be uncommented.